### PR TITLE
support for lancedb as vectordb

### PR DESCRIPTION
**Pull request by akashAD98**
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Added support for  lancedb as vector store

### CODE REVIEW

**Comment by akashAD98** [comment]
@dartpain can you please check it

File: application/core/settings.py
Comment (dartpain) [inline application/core/settings.py:1]: Im not sure if in memory is the best approach here as we have different containers that need to access lancedb. Does it work when you upload docs?

File: application/vectorstore/lancedb.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                pa.field("vector", pa.list_(pa.float32(), list_size=len(vectors[0]["vector"]))),
+                pa.field("text", pa.string()),
+                pa.field("metadata", pa.map_(pa.string(), pa.string())),
+            ])
+            self.table = self.db.create_table(self.table_name, schema=schema)
+
+        # Add data to the table
+        self.table.add(vectors)
+
+    def search(self, query: str, k: int = 2, *args, **kwargs):
```
Comment (dartpain) [inline application/vectorstore/lancedb.py:52]: Please make sure that you filter documents using self._path

File: application/vectorstore/lancedb.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
+
+
+
+from application.vectorstore.base import BaseVectorStore
+from application.core.settings import settings
+import lancedb
```
Comment (dartpain) [inline application/vectorstore/lancedb.py:6]: Lets lazy load it inside __init__

**Review by dartpain — CHANGES_REQUESTED** [review]
Thank you so much for creating this PR. really appreciate that its using native library too!
Requested a few minor change and happy to merge after.

File: application/vectorstore/lancedb.py (hunk 1 above)
Comment (akashAD98) [reply application/vectorstore/lancedb.py:62]: yes please check it

**Comment by akashAD98** [comment]
i created new pr so closing this

**Pull request closed without merging**